scrapbook/models.py

```diff
@@ -0,0 +1,312 @@
+        """list: a list of cell execution timings in cell order"""
+        return [
+            # TODO: Other timing conventions?
+            cell.metadata.get('papermill', {}).get('duration', 0.0)
+                if cell.get("execution_count") else None
+            for cell in self.node.cells
+        ]
+
+    @property
+    def execution_counts(self):
+        """list: a list of cell execution counts in cell order"""
+        return [
+            cell.get("execution_count") for cell in self.node.cells
+        ]
+
+    @property
+    def papermill_metrics(self):
+        """pandas dataframe: dataframe of cell execution counts and times"""
+        df = pd.DataFrame(columns=['filename', 'cell', 'value', 'type'])
+
+        for i, cell in enumerate(self.node.cells):
+            execution_count = cell.get("execution_count")
+            if not execution_count:
+                continue
+            name = "Out [{}]".format(str(execution_count))
+            value = cell.metadata.get('papermill', {}).get('duration', 0.0)
+            df.loc[i] = self.filename, name, value, "time (s)"
+        return df
+
+    @property
+    def parameter_dataframe(self):
+        """pandas dataframe: dataframe of notebook parameters"""
+        # Meant for backwards compatibility to papermill's dataframe method
+        return pd.DataFrame(
+            [[name, self.parameters[name], 'parameter', self.filename]
+                for name in sorted(self.parameters.keys())],
+            columns=['name', 'value', 'type', 'filename'])
+
+    @property
+    def scrap_dataframe(self):
+        """pandas dataframe: dataframe of cell scraps"""
+        # Meant for backwards compatibility to papermill's dataframe method
+        return pd.DataFrame(
+            [[name, self.scraps[name], 'record', self.filename]
+                for name in sorted(self.scraps.keys())],
+            columns=['name', 'value', 'type', 'filename'])
+
+    @property
+    def papermill_dataframe(self):
+        """pandas dataframe: dataframe of notebook parameters and cell scraps"""
+        # Meant for backwards compatibility to papermill's dataframe method
+        return self.parameter_dataframe.append(self.scrap_dataframe, ignore_index=True)
+
+    def _fetch_frames(self):
+        outputs = collections.OrderedDict()
+        for cell in self.node.cells:
+            for output in cell.get('outputs', []):
+                if 'scrapbook' in output.get('metadata', {}):
+                    output_name = output.metadata.scrapbook.get('name')
+                    if output_name:
+                        outputs[output_name] = output
+                # Backwards compatibility
+                if 'papermill' in output.get('metadata', {}):
+                    output_name = output.metadata.papermill.get('name')
+                    if output_name:
+                        outputs[output_name] = output
+        return outputs
+
+    @property
+    def frames(self):
```

Comment:
Perhaps rename frames since it is close to dataframes.

Reply:
`photo`? `snap`?

Reply:
Ideas, past and present, with my thoughts on the wording choice. Not many ones I liked tbh, though my wife just suggested highlight which might work and doesn't overlap?

`frame(s)` -- thematic but maybe not clear for non-native speakers / close to dataframes?
`display(s)` -- ok, not clear that it's presisting? Overlaps with ipython display which is good and bad
`show(???)` -- too vague?
`photo(s)` -- implies only images
`snap(s)` -- I'd think this was a checkpointing call
`highlight(s)` -- I think I like this one
`bejewel(s)` -- funny but probably not the best choice
`album` -- again implies images only
`cover(s)` -- some wording ambiguity still
`post(s)` -- too RESTy?
`print` -- overloaded with python print? metaphor also doesn't quite work there

Reply:
highlight works for me.